asserts/snap_asserts.go

```diff
@@ -365,7 +365,7 @@ func (validation *Validation) checkConsistency(db RODatabase, acck *AccountKey)
 		"snap-id": validation.ApprovedSnapID(),
 	})
 	if err == ErrNotFound {
-		return fmt.Errorf("validation assertion for snap-id %q does not have a matching snap-declaration assertion", validation.SnapID())
+		return fmt.Errorf("validation assertion for snap-id %q does not have a matching snap-declaration assertion for it", validation.ApprovedSnapID())
```

Comment:
I see & like what you're trying to do with the "for" v "by" in these error messages but it seems a bit subtle. Maybe this message should be "validation assertion for snap-id {{snap-id}} does not have a matching snap-declaration assertion for approved-snap-id {{approved-snap-id}}"?